submissions/ORybka/dom-js/main.js

```diff
@@ -0,0 +1,222 @@
+    mapPosition: ['scale(1.1)', '-40%', '-10%'],
+    dead: [1, 1, 1],
+  },
+  {
+    id: 8,
+    name: 'martell',
+    slogan: '"Unbowed, Unbent, Unbroken"',
+    description: "House Martell of Sunspear is one of the Great Houses of Westeros. It rules the peninsula of Dorne in the far south of the continent from their castle Sunspear. Though loyal to the Iron Throne, the Martells were never conquered by the Targaryens and pursued a more isolated role in wider political events since Robert's Rebellion.",
+    mapPosition: ['scale(1.1)', '20%', '-20%'],
+    dead: [1, 1, 1],
+  },
+];
+
+let button;
+let buttonLi;
+let image;
+let imageId;
+let imageCollection = [];
+let previousId = 0;
+const number = 3;
+const time = 1450;
+const houseElement = document.querySelector('.house');
+const shieldElement = document.querySelector('.shield');
+const sloganElement = document.querySelector('.slogan');
+const descriptionElement = document.querySelector('.description');
+const background = document.querySelector('.content-container');
+const menu = document.querySelector('.menu-container');
+const navbar = document.querySelector('.navbar');
+const images = document.querySelector('.content-img-container');
+const hiddenIcon = document.querySelector('.icon');
+const header = document.querySelector('.header-container');
+
+function createList() {
+  houses.forEach(({ id, name }) => {
+    buttonLi = document.createElement('li');
+    button = document.createElement('button');
+    button.className = 'menu-item';
+    button.id = id;
+    button.innerHTML = `
+    <img class="menu-item-img" src="assets/shields/${name}.png"><span>house ${name}</span>`;
+    buttonLi.appendChild(button);
+    menu.appendChild(buttonLi);
+  });
+}
+
+function changeContent(e) {
+  let target = e.target;
+  if (!target.matches('button')) {
+    target = target.parentNode;
+  }
+  const btn = target.closest('.menu-item');
+  if (btn) {
+    id = target.getAttribute('id');
+    createElements(houses[id]);
+    addImages(houses[id]);
+    animateMap(houses[id]);
+  }
+}
+
+menu.addEventListener('click', changeContent);
+
+function createElements({ name, slogan, description }) {
+  houseElement.innerText = `house ${name}`;
+  shieldElement.setAttribute('src', `assets/shields/${name}.png`);
+  sloganElement.innerText = `${slogan}`;
+  descriptionElement.innerHTML = `<span>${description}</span>`;
+  background.style.backgroundColor = `var(--${name})`;
+}
+
+function addImages(item) {
+  if (images.children.length === 0) {
+    createImage(item);
+  } else {
+    images.innerHTML = '';
+    createImage(item);
+  }
+}
+
+function createImage({ name }) {
+  for (let i = 0; i < number; i++) {
+    image = document.createElement('div');
+    image.className = 'content-img';
+    image.id = i;
+    images.appendChild(image);
```

Comment:
You can use the `documentFragment` for adding only one node to the `images` node. More info [here](https://developer.mozilla.org/ru/docs/Web/API/DocumentFragment).
It will help you to increase the performance of your app, because DOM operations cost a lot